temporalio/contrib/openai_agents/temporal_openai_agents.py

```diff
@@ -267,6 +268,93 @@ async def run_activity(ctx: RunContextWrapper[Any], input: str) -> Any:
             strict_json_schema=True,
         )
 
+    @classmethod
+    def nexus_operation_as_tool(
+        cls,
+        operation: nexusrpc.Operation[Any, Any],
+        *,
+        service: Type[Any],
+        endpoint: str,
+        schedule_to_close_timeout: Optional[timedelta] = None,
+    ) -> Tool:
+        """Convert a Nexus operation into an OpenAI agent tool.
+
+        .. warning::
+            This API is experimental and may change in future versions.
+            Use with caution in production environments.
+
+        This function takes a Nexus operation and converts it into an
+        OpenAI agent tool that can be used by the agent to execute the operation
+        during workflow execution. The tool will automatically handle the conversion
+        of inputs and outputs between the agent and the operation.
+
+        Args:
+            fn: A Nexus operation to convert into a tool.
+            service: The Nexus service class that contains the operation.
+            endpoint: The Nexus endpoint to use for the operation.
+
+        Returns:
+            An OpenAI agent tool that wraps the provided operation.
+
+        Example:
+            >>> @nexusrpc.service
+            ... class WeatherService:
+            ...     get_weather_object_nexus_operation: nexusrpc.Operation[WeatherInput, Weather]
+            >>>
+            >>> # Create tool with custom activity options
+            >>> tool = nexus_operation_as_tool(
+            ...     WeatherService.get_weather_object_nexus_operation,
+            ...     service=WeatherService,
+            ...     endpoint="weather-service",
+            ... )
+            >>> # Use tool with an OpenAI agent
+        """
+
+        def operation_callable(input):
+            raise NotImplementedError("This function definition is used as a type only")
+
+        operation_callable.__annotations__ = {
+            "input": operation.input_type,
+            "return": operation.output_type,
+        }
+        operation_callable.__name__ = operation.name
+
+        schema = function_schema(operation_callable)
```

Comment:
We support activities having access to an agent context as the first param but I am concerned Nexus operation limitations of single param forbid having context as the first param and input as the second. Suggestions? I think we may just have to document for now that tools backed by Nexus operations can't have access to context at this time.

Reply:
I think that is ok since we provide `nexus_operation_as_tool` as a convenience method. You should still be able to define a regular function tool, grab what you need from the context, and invoke the Nexus operation from there.

Reply:
Agreed behavior is ok, it's just the optics are confusing/inconsistent. "So, in-workflow tools can have mutable context, activity tools can have immutable context, and nexus tools can have no context?" We need to document this at least I think.